### ISSUE TITLE
Calc property calculate is not correct!

### ISSUE THREAD

**Issue by mqliutie** [webpack-contrib/less-loader#107]
When use command line  it is correct.
But use less-loader is not correct !

``` CSS
width: ~"calc(80% - 36px)";
```

result is `width : 44%`; I want `width:calc(80% - 36px)`

**Comment by zz-zhangzhao** [comment]
@mqliutie Have you solved it?

I met the same problem.

**Comment by dyf19118** [comment]
i met the same problem too.
Before extracting css styles that compiled from less into separated css files, it works well.
`max-height: calc(~'100vh - 200px');`
which compiles to `calc(100vh - 200px)`.
But after extracting, it was compiled to calc(-100vh), which leads to unexpected result.
Will this be solved, or i must avoid using calc as a compromise.
@TheLarkInn

**Comment by dyf19118** [comment]
@TheLarkInn I'm sorry for pointing out this issue without checking the reason behind such situation.

I found the problem i pointed out was caused by the default behavior of the directive `@import` of less, which will recompile the css file(extracted with less & css loader) imported.

The origin code `max-height: calc(~'100vh - 200px');` was first compiled to `max-height: calc(100vh - 200px);` by less-loader, it's okay.

Then i import the extracted css file in my less file, it was recompiled to `max-height: calc(-100vh);` , which leads to the unexpected result.

Solution to solve this problem: `@import (inline) 'extracted_css_file.css';`  It will prevent the default compile behavior.

@zz-zhangzhao For your situation, you can write like this `width: calc(~"80% - 36px")` instead.

**Comment by fireswork** [comment]
i have this problem ,too.
height: calc(100% - 110px);
result: -10%..............

**Comment by dyf19118** [comment]
@fireswork please check the comment i made before and you'll know the solution.

**Comment by henrikhelmers** [comment]
Could this solution also work for less-loader? Having to escape all calc operations makes the code harder to read.

https://github.com/less/less.js/commit/a48c24c4dd3c13e00a20ece80323768496a96b36

**Comment by originlondon-dev-transfer** [comment]
That isn't a solution, really. It is a work around. This needs to work- calc() is fairly central to css. It's not an obscure hack.

**Comment by originlondon-dev-transfer** [comment]
I aim was to use calc( 100% - @sidebar-width ); 
Tried all variants of using the literal:
 calc( ~"100% - @{sidebar-width}" ); 
~"calc( 100% - @{sidebar-width} )";
calc(100% ~"- @{sidebar-width}"); 
etc etc.
None worked
One compilation error said that the variable was not defined. Defining it in the element worked, then failed - no idea why. 

I tried defining a :root custom property's value  with a LESS variable. Nooooo.

I tried using the calc() in an @import (inline) "unlessable.css"; Not using a variable, as this solution suggests this compiles incorrectly too - with calc( 100% - 225px ) being brought in as -125%, even when using @import (inline)

But, unbelievably, putting the LESS variable in the css element definition in the css file imported by @import (inline) then using this worked!
 calc( ~"100% - @{sidebar-width}" ); 

So 
@import (inline) "unlessable.css"
and in unlessable.css

do
.element {
   @sidebar-width: 225px;
    max-width: calc( ~"100% - @{sidebar-width}" );
}
This works in a css file!
but
.element {
max-width: calc( 100% - 225px);
}
compiles to 
max-width: calc( -125px);

This should really be fixed.
I will see if I can take a look if I have time....
